hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/net/NetworkTopology.java

```diff
@@ -539,10 +538,12 @@ protected Node chooseRandom(final String scope, String excludedScope,
         netlock.readLock().unlock();
       }
     }
-    LOG.debug("Choosing random from {} available nodes on node {},"
-        + " scope={}, excludedScope={}, excludeNodes={}. numOfDatanodes={}.",
-        availableNodes, innerNode, scope, excludedScope, excludedNodes,
-        numOfDatanodes);
+    if (LOG.isDebugEnabled()) {
```

Comment:
Here

Reply:
There is string concatenation here, so we need to keep this judgment.

Reply:
Reasonable.